builtin/providers/aws/resource_aws_api_gateway_deployment_test.go

```diff
@@ -0,0 +1,169 @@
+					testAccCheckAWSAPIGatewayDeploymentAttributes(&conf),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSAPIGatewayDeploymentAttributes(conf *apigateway.Deployment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if *conf.Id == "" {
+			return fmt.Errorf("empty ID")
+		}
+		if *conf.Description != "This is a test" {
+			return fmt.Errorf("wrong Description")
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAPIGatewayDeploymentExists(n string, res *apigateway.Deployment) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No API Gateway Deployment ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+		req := &apigateway.GetDeploymentInput{
+			DeploymentId: aws.String(rs.Primary.ID),
+			RestApiId:    aws.String(s.RootModule().Resources["aws_api_gateway_rest_api.test"].Primary.ID),
+		}
+		describe, err := conn.GetDeployment(req)
+		if err != nil {
+			return err
+		}
+
+		if *describe.Id != rs.Primary.ID {
+			return fmt.Errorf("APIGateway Deployment not found")
+		}
+
+		res.Id = describe.Id
+		res.Description = describe.Description
+
+		return nil
+	}
+}
+
+func testAccCheckAWSAPIGatewayDeploymentDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).apigateway
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_api_gateway_resource" {
+			continue
+		}
+
+		req := &apigateway.GetDeploymentsInput{
+			RestApiId: aws.String(s.RootModule().Resources["aws_api_gateway_rest_api.test"].Primary.ID),
+		}
+		describe, err := conn.GetDeployments(req)
+
+		if err == nil {
+			if len(describe.Items) != 0 &&
+				*describe.Items[0].Id == rs.Primary.ID {
+				return fmt.Errorf("API Gateway Deployment still exists")
+			}
+		}
+
+		aws2err, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if aws2err.Code() != "NotFoundException" {
+			return err
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+const testAccAWSAPIGatewayDeploymentConfig = `
+resource "aws_api_gateway_rest_api" "test" {
+  name = "test"
+}
+
+resource "aws_api_gateway_resource" "test" {
+  api_id = "${aws_api_gateway_rest_api.test.id}"
+  parent_resource_id = "${aws_api_gateway_rest_api.test.root_resource_id}"
+  path_part = "test"
+}
+
+resource "aws_api_gateway_method" "test" {
+  api_id = "${aws_api_gateway_rest_api.test.id}"
+  resource_id = "${aws_api_gateway_resource.test.id}"
+  http_method = "GET"
+  authorization = "NONE"
+}
+
+resource "aws_api_gateway_method_response" "error" {
+  api_id = "${aws_api_gateway_rest_api.test.id}"
+  resource_id = "${aws_api_gateway_resource.test.id}"
+  http_method = "${aws_api_gateway_method.test.http_method}"
+  status_code = "400"
+}
+
+resource "aws_api_gateway_integration" "test" {
+  depends_on = ["aws_api_gateway_method.test"]
```

Comment:
This line shouldn't be necessary, the relationship is already defined 3 lines below - `http_method = "${aws_api_gateway_method.test.http_method}"`.